### PR TITLE
[FIX] Fixes #303 by adding the header a suitable args to the test compile

### LINKED ISSUE #303: Building on Python 3.9 / macOS 10.15.6 fails without -Wno-error=implicit-function-declaration
As the title says, building on macOS 10.15.6, Python 3.9 (from Homebrew), fails since there's somehow an implicit `-Werror` in the clang command line that makes clang barf when no headers are used for finding the symbol(s).

```
$ pip list
Package    Version
---------- -------
Cython     0.29.21
numpy      1.20.0
pip        20.3.1
setuptools 51.0.0
wheel      0.36.1
```

```
$ LDFLAGS="-L/usr/local/lib" CFLAGS="-I/usr/local/include -Wno-error" python setup.py -v build_ext

running build_ext
Link FFTW dynamically
Compiler include_dirs: ['/usr/local/include', '/usr/local/opt/openssl@1.1/include', '/usr/local/opt/sqlite/include', '/Users/akx/envs/pyFFTW/include', '/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/include/python3.9']
3.9.0 (default, Dec  3 2020, 16:09:02)
[Clang 12.0.0 (clang-1200.0.32.27)]
objects: []
libraries: []
include dirs: ['/Users/akx/build/pyFFTW/include', '/Users/akx/build/pyFFTW/pyfftw', '/Users/akx/envs/pyFFTW/lib/python3.9/site-packages/numpy/core/include', '/Users/akx/envs/pyFFTW/include']
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -I/usr/local/include -Wno-error -I/Users/akx/build/pyFFTW/include -I/Users/akx/build/pyFFTW/pyfftw -I/Users/akx/envs/pyFFTW/lib/python3.9/site-packages/numpy/core/include -I/Users/akx/envs/pyFFTW/include -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/Users/akx/envs/pyFFTW/include -I/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-xs9trp4v/None.c -o /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-xs9trp4v/None.o


clang /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-xs9trp4v/None.o -L/Users/akx/envs/pyFFTW/lib -L/usr/local/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/sqlite/lib -o /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-xs9trp4v/a.out


Checking with includes ['fftw3.h']...ok
objects: []
libraries: ['fftw3']
include dirs: ['/Users/akx/build/pyFFTW/include', '/Users/akx/build/pyFFTW/pyfftw', '/Users/akx/envs/pyFFTW/lib/python3.9/site-packages/numpy/core/include', '/Users/akx/envs/pyFFTW/include']
clang (... as above ...) -c /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-duahkw59/fftw_plan_dft.c -o /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-duahkw59/fftw_plan_dft.o

/var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-duahkw59/fftw_plan_dft.c:2:17: error: implicit declaration of function 'fftw_plan_dft' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                fftw_plan_dft();
                ^
1 error generated.

Checking for fftw_plan_dft...no
objects: []
libraries: ['fftw3f']
include dirs: ['/Users/akx/build/pyFFTW/include', '/Users/akx/build/pyFFTW/pyfftw', '/Users/akx/envs/pyFFTW/lib/python3.9/site-packages/numpy/core/include', '/Users/akx/envs/pyFFTW/include']
clang (... as above ...) -c /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-juiekt3y/fftwf_plan_dft.c -o /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-juiekt3y/fftwf_plan_dft.o

/var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-juiekt3y/fftwf_plan_dft.c:2:17: error: implicit declaration of function 'fftwf_plan_dft' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                fftwf_plan_dft();
                ^
1 error generated.

Checking for fftwf_plan_dft...no
objects: []
libraries: ['fftw3l']
include dirs: ['/Users/akx/build/pyFFTW/include', '/Users/akx/build/pyFFTW/pyfftw', '/Users/akx/envs/pyFFTW/lib/python3.9/site-packages/numpy/core/include', '/Users/akx/envs/pyFFTW/include']
clang (... as above ...) -c /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-rqqq1nam/fftwl_plan_dft.c -o /var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-rqqq1nam/fftwl_plan_dft.o

/var/folders/n0/8r8c6bms2fj7753rvhmtybqm0000gn/T/pyfftw-rqqq1nam/fftwl_plan_dft.c:2:17: error: implicit declaration of function 'fftwl_plan_dft' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                fftwl_plan_dft();
                ^
1 error generated.

Checking for fftwl_plan_dft...no
{'HAVE_DOUBLE': False, 'HAVE_DOUBLE_OMP': False, 'HAVE_DOUBLE_THREADS': False, 'HAVE_DOUBLE_MULTITHREADING': False, 'HAVE_DOUBLE_MPI': False, 'HAVE_SINGLE': False, 'HAVE_SINGLE_OMP': False, 'HAVE_SINGLE_THREADS': False, 'HAVE_SINGLE_MULTITHREADING': False, 'HAVE_SINGLE_MPI': False, 'HAVE_LONG': False, 'HAVE_LONG_OMP': False, 'HAVE_LONG_THREADS': False, 'HAVE_LONG_MULTITHREADING': False, 'HAVE_LONG_MPI': False, 'HAVE_MPI': False}
error: Could not find any of the FFTW libraries
```

# Workaround

The error can be worked around with `-Wno-error=implicit-function-declaration` in CFLAGS, so

```
LDFLAGS="-L/usr/local/lib" CFLAGS="-I/usr/local/include -Wno-error=implicit-function-declaration" python setup.py -v build_ext
```

successfully builds the extension.

Similarly, to `pip install` the extension so long as no wheels exist yet (#301),

```
pip install Cython
LDFLAGS="-L/usr/local/lib" CFLAGS="-I/usr/local/include -Wno-error=implicit-function-declaration" pip install pyfftw
```
does the trick.

(As an aside, if you get a problem referencing Mecab and  NLP framework and you're using DYLD_LIBRARY_PATH in your command line, try `brew unlink mecab`...)

If you need any further details, let me know, naturally 😄

### PR DESCRIPTION
**Pull request by hgomersall**

### CODE REVIEW

**Review by JeppeKlitgaard — CHANGES_REQUESTED** [review]
I haven't actually run this. I think we should also check CI output, which I am happy to do at a later point

File: setup.py [hunk 1]
```diff
@@ -403,13 +409,14 @@ def has_function(self, function, includes=None, objects=None, libraries=None,
             try:
                 fname = os.path.join(tmpdir, '%s.c' % function)
                 f = open(fname, 'w')
+                print(includes)
```
[the remark this answers is not in the document]
Comment (hgomersall) [reply setup.py]: whoops yes

File: setup.py [hunk 2]
```diff
@@ -403,13 +409,14 @@ def has_function(self, function, includes=None, objects=None, libraries=None,
...
                 for inc in includes:
                     f.write('#include <%s>\n' % inc)
                 f.write("""\
                 int main() {
                 """)
                 if function:
-                    f.write('%s();\n' % function)
+                    f.write('%s(%s);\n' % (function, function_args))
```
[the remark this answers is not in the document]
Comment (hgomersall) [reply setup.py]: Fixed in latest push (including all the rest)

File: setup.py [hunk 3]
```diff
@@ -392,31 +392,30 @@ def has_function(self, function, includes=None, objects=None, libraries=None,
...
         if includes:
             msg += " with includes " + str(includes)
         msg += "..."
         status = "no"
 
-        log.debug("objects: %s" % objects)
-        log.debug("libraries: %s" % libraries)
-        log.debug("include dirs: %s" % include_dirs)
+        log.debug(f"objects: {objects}")
+        log.debug(f"libraries: {libraries}")
```
Comment (JeppeKlitgaard) [inline setup.py:402]: No need to change this here at all (as in, this is extraordinarily nitty), but perhaps you find this oddity interesting: https://docs.python.org/3/howto/logging.html#optimization

TL;DR: interpolation in logging statements is usually done with `'%s'` style and passing in arguments to the log function.

Again, only mentioned because I find the case interesting – no need to change here

**Review by JeppeKlitgaard — APPROVED** [review]

**Comment by JeppeKlitgaard** [comment]
I haven't actually run the code, but it seems to pass CI, so LGTM 🚀
